subprojects/snapshots/src/integTest/groovy/org/gradle/internal/vfs/VirtualFileSystemRetentionIntegrationTest.groovy

```diff
@@ -214,6 +214,43 @@ class VirtualFileSystemRetentionIntegrationTest extends AbstractIntegrationSpec
         executedAndNotSkipped ":compileJava", ":classes", ":run"
     }
 
+    def "detects input file change just before the task is executed"() {
```

Comment:
Is this the right approach, using a task to make the change? I mean, we can do it like this, but this kind of usage will be flagged (hopefully soon), and we'll need to do the change outside the build anyway, no?

Reply:
We can never flag a task making changes to the primary inputs of the build, since those changes also can come from the end user. I'll try to find another solution.

Reply:
I am now using the blocking http server, so we stop the build before/after the task and make the changes from the test thread.

Reply:
Great solution!